BHoM_UI/Components/oM/CreateObject2.cs

```diff
@@ -0,0 +1,147 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.UI.Templates;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using BH.oM.Base;
+using BH.oM.UI;
+using BH.Engine.Reflection;
+using BH.Engine.Serialiser;
+using System.Collections;
+using BH.Engine.UI;
+
+namespace BH.UI.Components
+{
+    public class CreateObject2Caller : MethodCaller
+    {
+        /*************************************/
+        /**** Properties                  ****/
+        /*************************************/
+
+        public override System.Drawing.Bitmap Icon_24x24 { get; protected set; } = Properties.Resources.CreateBHoM;
+
+        public override Guid Id { get; protected set; } = new Guid("D1B418EF-4CBB-4A7E-AC73-67FA78453F1A");
+
+        public override string Name { get; protected set; } = "CreateObject2";
+
+        public override string Category { get; protected set; } = "oM";
+
+        public override string Description { get; protected set; } = "Creates an instance of a selected type of BHoM object";
+
+
+        /*************************************/
+        /**** Constructors                ****/
+        /*************************************/
+
+        public CreateObject2Caller() : base()
+        {
+            List<MemberInfo> possibleItems = new List<MemberInfo>();
+            possibleItems.AddRange(Engine.Reflection.Query.BHoMTypeList().Where(t => t?.GetInterface("IImmutable") == null));
+            possibleItems.AddRange(Engine.UI.Query.CreateItems());
```

Comment:
All the methods that provide the possible items are in the [SearchItem](https://github.com/BHoM/BHoM_UI/blob/master/UI_Engine/Query/SearchItems.cs) method. Would it be convenient to move this logic there as well?

This would keep the list of possible items consistent between the global menu and the local menus.
https://github.com/BHoM/BHoM_UI/blob/master/BHoM_UI/Global/SearchMenu.cs#L84

Reply:
Possibly. I would keep that for another PR though as none of the current components are currently going that route.